float64[3] data          # actual matrix data, size is (step * rows)


